src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -972,6 +972,48 @@ public async Task Alpn_H3_Success()
             await connection.DisposeAsync();
         }
 
+        [Fact]
+        public async Task H3_AltUsedHeader()
+        {
+            var options = new Http3Options() { Alpn = SslApplicationProtocol.Http3.ToString() };
+            using Http3LoopbackServer server = CreateHttp3LoopbackServer(options);
+
+            Http3LoopbackConnection connection = null;
+            HttpRequestData requestData = null;
+            Task serverTask = Task.Run(async () =>
+            {
+                connection = (Http3LoopbackConnection)await server.EstablishGenericConnectionAsync();
+                requestData = await connection.ReadRequestDataAsync(readBody: false);
```

Comment:
`HandleRequestAsync` also returns request data, so you don't have to send the response separately on the following lines.